t/t0100-previous.sh

```diff
@@ -50,6 +72,17 @@ test_expect_success 'merge @{-1}~1' '
 	grep "Merge branch '\''other'\''" actual
 '
 
+test_expect_success 'merge -~1' '
```

Comment:
It would make a lot of sense to discuss supporting `-~1` on the Git mailing list first, before putting a lot of work into a patch that is likely to be rejected. My understanding is that we do _not_ want to support this, it is too ambiguous to treat `-<some-random-text>` as having anything to do with the previous branch.

Reply:
It's certainly been a bit of work. I'm sorry it wasn't accepted, even just '-', you also spent some time on this.
For me it has been interesting and enriching, so all is not lost :-)
Thanks.